src/ripple/peerfinder/impl/SlotImp.h

```diff
@@ -146,6 +160,9 @@ class SlotImp : public Slot
     beast::IP::Endpoint m_remote_endpoint;
     boost::optional <beast::IP::Endpoint> m_local_endpoint;
     boost::optional <RipplePublicKey> m_public_key;
+    // ports are uint16_t values, but we use uint32_t so we can
```

Comment:
Consider

    static std::int32_t const noPort = -1;
    std::atomic <std::int32_t> m_listening_port = noPort;

and then earlier:

    if (value == noPort)